### PR TITLE
Integrate "What's new" table generation into doc scripts

### PR DESCRIPTION
**Pull request by AlexJerabek**
This PR plugs the whats-new script into GenerateDocs.cmd. Phase 2 will be to reference those .md files from the -pr repo.

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[deb290a](https://github.com/OfficeDev/office-js-docs-reference/commits/deb290aaef0a66c0cf8e204268bad5c114384270)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/requirement-set-tables/excel-1_1.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_1.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_10.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_10.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_11.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_11.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_12.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_12.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_2.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_2.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_3.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_3.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_4.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_4.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_5.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_5.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_6.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_6.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_7.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_7.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_8.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_8.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_9.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_9.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-online.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-online.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-preview.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-preview.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_1.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_1.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_2.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_2.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_3.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_3.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_4.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_4.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_5.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_5.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_6.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_6.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_7.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_7.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_8.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_8.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-preview.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-preview.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/powerpoint-preview.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/powerpoint-preview.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/word-1_1.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/word-1_1.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/word-1_2.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/word-1_2.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/word-1_3.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/word-1_3.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/word-preview.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/word-preview.md) | :white_check_mark:Succeeded | |
[generate-docs/GenerateDocs.cmd](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/generate-docs/GenerateDocs.cmd) | :white_check_mark:Succeeded | |
[generate-docs/tools/dts-utilities.ts](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/generate-docs/tools/dts-utilities.ts) | :white_check_mark:Succeeded | |
[generate-docs/tools/version-remover.ts](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/generate-docs/tools/version-remover.ts) | :white_check_mark:Succeeded | |
[generate-docs/tools/whats-new.ts](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/generate-docs/tools/whats-new.ts) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C11%5C6%5C0e1871f2-88cb-c16b-6d81-dc80e4f66cef%5CPullRequest%5C202011060115309192-828%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=K6eTYS8VYHVe%2FcFzOYmJCMKSzKYwv9U3rj4TjC1jsQQ%3D&st=2020-11-06T01%3A55%3A50Z&se=2020-12-07T02%3A00%3A50Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/0e1871f2-88cb-c16b-6d81-dc80e4f66cef?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

File: docs/requirement-set-tables/outlook-1_8.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+| Class | Fields | Description |
+|:---|:---|:---|
+|[AppointmentCompose](/javascript/api/outlook/outlook.appointmentcompose)|[addFileAttachmentFromBase64Async(base64File: string, attachmentName: string, options?: CommonAPI.AsyncContextOptions &  { isInline: boolean }, callback?: (asyncResult: CommonAPI.AsyncResult<string>) => void)](/javascript/api/outlook/outlook.appointmentcompose#addfileattachmentfrombase64async-base64file--attachmentname--options--isinline--callback--asyncresult-)|Adds a file to a message or appointment as an attachment.|
```
Comment (ElizabethSamuel-MSFT) [inline docs/requirement-set-tables/outlook-1_8.md:3]: Seems like we should clean up CommonAPI refs

File: docs/requirement-set-tables/excel-preview.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+||[type](/javascript/api/excel/excel.linkeddatatypeaddedeventargs#type)|Gets the type of the event. See Excel.EventType for details.|
+|[LinkedDataTypeCollection](/javascript/api/excel/excel.linkeddatatypecollection)|[getCount()](/javascript/api/excel/excel.linkeddatatypecollection#getcount--)|Gets the number of linked data types in the collection.|
+||[getItem(key: number)](/javascript/api/excel/excel.linkeddatatypecollection#getitem-key-)|Gets a linked data type by service id.|
+||[getItemAt(index: number)](/javascript/api/excel/excel.linkeddatatypecollection#getitemat-index-)|Gets a linked data type by its index in the collection.|
+||[getItemOrNullObject(key: number)](/javascript/api/excel/excel.linkeddatatypecollection#getitemornullobject-key-)|Gets a linked data type by ID. If the linked data type does not exist, an object with its `isNullObject` property set to `true`. For further information, see {@link https://docs.microsoft.com/office/dev/add-ins/develop/application-specific-api-model#ornullobject-methods-and-properties | *OrNullObject methods and properties}.|
```
Comment (ElizabethSamuel-MSFT) [inline docs/requirement-set-tables/excel-preview.md:18]: The `{@link...` notation - convert to regular link?

**Review by ElizabethSamuel-MSFT — COMMENTED** [review]
This is great! Spot checked the content and noticed a couple things.

File: docs/requirement-set-tables/outlook-1_8.md (hunk 1 above)
Comment (AlexJerabek) [reply docs/requirement-set-tables/outlook-1_8.md:3]: Good catch. I'll add that logic to the post-processor.

File: docs/requirement-set-tables/excel-preview.md (hunk 2 above)
Comment (AlexJerabek) [reply docs/requirement-set-tables/excel-preview.md:18]: I'm hesitant to edit the ref doc content too much. That link should display correctly. That said, I thought the tool was stripping out text that is that long, so I'll investigate that.

**Comment by AlexJerabek** [comment]
Thanks for the review @ElizabethSamuel-MSFT. I've fixed the Outlook references to `CommonAPI`. I've also tweaked the "first sentence" parsing for the descriptions. The tool should trim them better now.

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

**Comment by o365devx** [comment]
Docs Build status updates of commit _[5831cae](https://github.com/OfficeDev/office-js-docs-reference/commits/5831cae83367ee54a5b1366b206771e1a1774ebb)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/requirement-set-tables/excel-1_1.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_1.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_10.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_10.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_11.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_11.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_12.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_12.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_2.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_2.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_3.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_3.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_4.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_4.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_5.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_5.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_6.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_6.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_7.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_7.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_8.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_8.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-1_9.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-1_9.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-online.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-online.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/excel-preview.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/excel-preview.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_1.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_1.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_2.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_2.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_3.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_3.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_4.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_4.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_5.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_5.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_6.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_6.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_7.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_7.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_8.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_8.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-1_9.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-1_9.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/outlook-preview.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/outlook-preview.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/powerpoint-preview.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/powerpoint-preview.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/word-1_1.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/word-1_1.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/word-1_2.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/word-1_2.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/word-1_3.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/word-1_3.md) | :white_check_mark:Succeeded | |
[docs/requirement-set-tables/word-preview.md](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/docs/requirement-set-tables/word-preview.md) | :white_check_mark:Succeeded | |
[generate-docs/GenerateDocs.cmd](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/generate-docs/GenerateDocs.cmd) | :white_check_mark:Succeeded | |
[generate-docs/scripts/midprocessor.ts](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/generate-docs/scripts/midprocessor.ts) | :white_check_mark:Succeeded | |
[generate-docs/tools/dts-utilities.ts](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/generate-docs/tools/dts-utilities.ts) | :white_check_mark:Succeeded | |
[generate-docs/tools/version-remover.ts](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/generate-docs/tools/version-remover.ts) | :white_check_mark:Succeeded | |
[generate-docs/tools/whats-new.ts](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-WhatsNew/generate-docs/tools/whats-new.ts) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C11%5C6%5C0e1871f2-88cb-c16b-6d81-dc80e4f66cef%5CPullRequest%5C202011061941276306-828%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=LSdCGPVMiT%2BNCHEklB7yYc7RssQRYnabfsFaZm1gvmQ%3D&st=2020-11-06T20%3A22%3A57Z&se=2020-12-07T20%3A27%3A57Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/0e1871f2-88cb-c16b-6d81-dc80e4f66cef?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

**Pull request merged**